src/spikeinterface/preprocessing/detect_bad_channels.py

```diff
@@ -4,8 +4,122 @@
 import numpy as np
 from typing import Literal
 
-from .filter import highpass_filter
-from ..core import get_random_data_chunks, order_channels_by_depth, BaseRecording
+from spikeinterface.core.core_tools import define_function_from_class
+from spikeinterface.preprocessing.filter import highpass_filter
+from spikeinterface.core import get_random_data_chunks, order_channels_by_depth, BaseRecording
+from spikeinterface.core.channelslice import ChannelSliceRecording
+
+from inspect import signature
+
+
+class RemoveBadChannelsRecording(ChannelSliceRecording):
+    """
+    Removes bad channels.
+
+    {}
+
+    Returns
+    -------
+    removed_bad_channels_recording : RemoveBadChannelsRecording
+        The recording with bad channels removed
+    """
+
+    params_doc = """Different methods are implemented:
```

Comment:
I would move this outside of the class and just call it `_bad_channel_detection_kwargs_doc`